File: solutions/julia/sum-of-multiples/2/sum-of-multiples.jl
function get_one_set(limit, factor)
    factor == 0 ? Set() : 
     Set(collect(factor:factor:limit-1))
end

function sum_of_multiples(limit, factors)
    s = Set(0)
    for f=factors 
                 s = union(s, get_one_set(limit,f))
    end
    sum(s)
end

File: solutions/cpp/sum-of-multiples/2/sum_of_multiples.cpp
#include "sum_of_multiples.h"

#include <set>

using namespace std;

namespace sum_of_multiples {

    int to(set<int> factors, int limit){
        set<int> s{0};
        int sum = 0;
        for(int f: factors){
            for(int m=f; m<limit && m!=0; m+=f){
                   if(s.find(m)==s.end()){ 
                       s.insert(m);
                       sum += m;
                   }
            }
        }
        return sum;
    }
    

}  // namespace sum_of_multiples
